tests/suite/graceful_recovery_test.go

```diff
@@ -100,8 +102,122 @@ var _ = Describe("Graceful Recovery test", Ordered, Label("functional", "gracefu
+		// FIXME(bjee19) remove Skip() when https://github.com/nginxinc/nginx-gateway-fabric/issues/1108 is completed.
+		Skip("Test currently fails due to this issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1108")
+		runRestartNodeTest(teaURL, coffeeURL, files, &ns, false)
+	})
 })
 
+func runRestartNodeTest(teaURL, coffeeURL string, files []string, ns *core.Namespace, drain bool) {
+	nodeNames, err := getNodeNames()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(nodeNames).To(HaveLen(1))
+
+	kindNodeName := nodeNames[0]
+
+	if portFwdPort != 0 {
+		close(portForwardStopCh)
+	}
+
+	if drain {
+		_, err := exec.Command(
+			"kubectl",
+			"drain",
+			kindNodeName,
+			"--ignore-daemonsets",
+			"--delete-local-data",
+		).CombinedOutput()
+		if err != nil {
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		_, err = exec.Command(
+			"kubectl",
+			"delete",
+			"node",
+			kindNodeName,
+		).CombinedOutput()
+		if err != nil {
+			Expect(err).ToNot(HaveOccurred())
+		}
+	}
+
+	containerOutput, err := exec.Command(
+		"docker",
+		"container",
+		"ls",
+	).CombinedOutput()
+	if err != nil {
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	var containerName string
+	for _, line := range strings.Split(string(containerOutput), "\n") {
+		for _, word := range strings.Split(line, " ") {
+			// This is a potential weak spot in the code where we rely on the container which NGF
+			// is running on to contain "control-plane" in the name and for no other container to have that either.
+			// This is currently working in our test framework may break in the future.
+			if strings.Contains(word, "control-plane") {
+				containerName = strings.TrimSpace(word)
+				break
+			}
+		}
+	}
+	Expect(containerName).ToNot(Equal(""))
+
+	_, err = exec.Command(
+		"docker",
+		"restart",
+		containerName,
+	).CombinedOutput()
+	if err != nil {
+		fmt.Println(fmt.Sprint(err.Error()))
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	// need to wait for docker container to restart and be running before polling for ready NGF Pods or else we will error
+	Eventually(
+		func() bool {
+			output, err := exec.Command(
+				"docker",
+				"container",
+				"inspect",
+				containerName,
+			).CombinedOutput()
+			return strings.Contains(string(output), "\"Running\": true") && err == nil
+		}).
+		WithTimeout(timeoutConfig.CreateTimeout).
+		WithPolling(500 * time.Millisecond).
+		Should(BeTrue())
+
+	var podNames []string
+	Eventually(
+		func() bool {
+			podNames, err = framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetStatusTimeout)
+			return len(podNames) == 1 && err == nil
+		}).
+		WithTimeout(timeoutConfig.CreateTimeout).
+		WithPolling(500 * time.Millisecond).
+		Should(BeTrue())
+	ngfPodName := podNames[0]
+	Expect(ngfPodName).ToNot(Equal(""))
+
+	if portFwdPort != 0 {
+		ports := []string{fmt.Sprintf("%d:80", ngfHTTPForwardedPort), fmt.Sprintf("%d:443", ngfHTTPSForwardedPort)}
+		portForwardStopCh = make(chan struct{})
```

Comment:
Do we need to close this channel or is that handled by the framework?

Reply:
I close it above near the start of `runRestartNodeTest` because if I don't, since the NGF pod goes offline, in our portforwarding function it keeps pinging the pod which just prints endless amounts of error messages.

Reply:
I mean now that we've restarted the port forwarding, do we need to stop it once the test is over? Or is that handled by the testing framework on suite cleanup?

Reply:
Ah I see, sorry about the misunderstanding. Yep it looks like when `teardown()` is called in the suite's cleanup, it closes the channel.